hazelcast-client/src/test/java/classloading/ThreadLeakClientTest.java

```diff
@@ -42,13 +55,69 @@ public void testThreadLeak() {
     @Test
     public void testThreadLeak_withoutCluster() {
         ClientConfig clientConfig = new ClientConfig();
-        clientConfig.getConnectionStrategyConfig()
-                .setAsyncStart(true);
-        clientConfig.getNetworkConfig()
-                .setConnectionAttemptLimit(Integer.MAX_VALUE);
+        clientConfig.getConnectionStrategyConfig().setAsyncStart(true);
+        clientConfig.getNetworkConfig().setConnectionAttemptLimit(Integer.MAX_VALUE);
 
         HazelcastInstance client = HazelcastClient.newHazelcastClient(clientConfig);
 
         client.shutdown();
     }
+
+    @Test (expected = IllegalStateException.class)
+    public void testThreadLeakWhenClientCanNotStart() {
+        HazelcastClient.newHazelcastClient();
+        Assert.fail();
+    }
+
+    @Test (expected = IllegalStateException.class)
+    public void testThreadLeakWhenClientCanNotStartDueToAuthenticationError() {
+        try {
+            Hazelcast.newHazelcastInstance();
```

Comment:
The AbstractThreadLeakTest extends from HazelcastTestSupport; if you create your HZ through the HazelcastTestSupport, then the instance is automatically terminated at the end of the test as part of the 'after' handling.

Reply:
Yes, that would be nice but for this test, it uses real network instances which I believe is not possible to manage with HazelcastTestSupport. Hence, I could not see how I can do this.